skin/src/main/java/org/researchstack/skin/AppPrefs.java

```diff
@@ -52,22 +56,37 @@ public static AppPrefs getInstance()
      *
      * @return time in milliseconds
      */
-    public long getAutoLockTime()
-    {
+    public long getAutoLockTime() {
         boolean isAutoLocked = prefs.getBoolean(SettingsFragment.KEY_AUTO_LOCK_ENABLED, true);
 
         String time = prefs.getString(SettingsFragment.KEY_AUTO_LOCK_TIME, "1");
         int autoLockMins = isAutoLocked ? Integer.parseInt(time) : 60 * 24 * 365;
         return autoLockMins * 60 * 1000;
     }
 
-    public void setSkippedOnboarding(boolean skipped)
-    {
+    public void addDataProviderError(@Nullable @DataProvider.Errors String dataProviderError) {
```

Comment:
Perhaps some explanation here in comments or method header on why a developer would want to save error strings?